compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1674,6 +1674,12 @@ public BType readType(int cpI) throws IOException {
                     if (Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
                         objectSymbol.flags |= Flags.ANONYMOUS;
                     }
+                    if (Symbols.isFlagOn(flags, Flags.DISTINCT)) {
+                        objectSymbol.flags |= Flags.DISTINCT;
+                    }
+                    if (Symbols.isFlagOn(flags, Flags.ISOLATED)) {
```

Comment:
Did we check how https://github.com/ballerina-platform/ballerina-lang/issues/40619#issuecomment-1609088407 worked as expected without this?

Reply:
Yes, At the node level isolated [flag](https://github.com/ballerina-platform/ballerina-lang/blob/d08328e9b726c32553ce6bf9a5f2c70412a44e86/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L1677) information was available. This information was missing in the symbol and this symbol is used in the semantic API.